### PR TITLE
[YUNIKORN-531] migrate to github action

### PR DESCRIPTION
**Pull request by yangwwei**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,36 +37,11 @@ install: true
...
-      - make test
-      - make lint
-      after_success: bash <(curl -s https://codecov.io/bash)
     - stage: publish docker image
       deploy:
         provider: script
         script: make push
         on:
           branch: master
           condition: $TRAVIS_EVENT_TYPE = cron
```
Comment (wilfred-s) [inline .travis.yml:46]: This needs to be replaced with a scheduled github action or something like it. We do need to be careful as this is one part of the [distribution guidelines](https://incubator.apache.org/guides/distribution.html#docker) which we are breaking:
```
on:
  schedule:
    - cron:  '0 0 * * *'
```

Comment (yangwwei) [reply .travis.yml:46]: I leave this as an open item for now, let the Travis job push the docker images until we figure out how to do that with github action. I think I've tried to do this github action before, but I met some token issue unsolved. That was a few months ago, we probably need to fix this in a second step.

Comment (wilfred-s) [reply .travis.yml:46]: The fact that we push the image means we break the distribution guidelines. The latest tag is pointing to unreleased code which based on the distribution guidelines point 6 we cannot do. We do not rely on the image for testing or other release work should we not turn it off until we have the solution worked out that complies with the distribution guidelines?

Comment (yangwwei) [reply .travis.yml:46]: I think we can safely remove it, even for the e2e test today, we do the build from scratch.
Let me remove this, we can revisit if we need to have auto-push in the feature.

**Comment by yangwwei** [comment]
The UT `TestValidateConfigMapWrongRequest` failure was because the error contains different messages on the box used to run github action (ubuntu). "Temp failure in name resolution" vs "no such host", Fixing it by simply verify the error is not nil, instead of detail messages.

**Review by wilfred-s — APPROVED** [review]
LGTM committing

**Pull request closed without merging**